python/tests/domains/test_qnn.py

```diff
@@ -54,6 +54,8 @@ def test_simpleObserveN_QNN():
     assert np.isclose(data[-1], 0.33752631)
 
 
+## [SKIP_TEST]
+@pytest.mark.skip(reason="RuntimeError: Could not pack argument...")
```

Comment:
Issue here is that `py_observe_async.cpp` is not running something like `args = validateInputArguments(args)` like the current bindings do. This function will find any `np.ndarray` with 1D shape and call `tolist()` on it to make it compatible with our MLIR code. Here's the diff for the fix 
```diff
diff --git a/python/runtime/cudaq/algorithms/py_observe_async.cpp b/python/runtime/cudaq/algorithms/py_observe_async.cpp
index 8b211eac2..f81bf24ff 100644
--- a/python/runtime/cudaq/algorithms/py_observe_async.cpp
+++ b/python/runtime/cudaq/algorithms/py_observe_async.cpp
@@ -34,7 +34,8 @@ void bindObserveAsync(py::module &mod) {
 
         auto &platform = cudaq::get_platform();
         auto *argData = new cudaq::OpaqueArguments();
-        cudaq::packArgs(*argData, args);
+        args = simplifiedValidateInputArguments(args);
+        cudaq::packArgs(*argData, args, [](OpaqueArguments&, py::object&) {return false;});
         auto kernelName = kernel.attr("name").cast<std::string>();
         auto kernelMod = kernel.attr("module").cast<MlirModule>();
 
diff --git a/python/utils/OpaqueArguments.h b/python/utils/OpaqueArguments.h
index e1d5e7ac8..f8af6fade 100644
--- a/python/utils/OpaqueArguments.h
+++ b/python/utils/OpaqueArguments.h
@@ -123,6 +123,38 @@ inline py::args validateInputArguments(kernel_builder<> &kernel,
   return processed;
 }
 
+/// @brief FIXME. This is a simple version of the above function, 
+// it will only process numpy arrays of 1D shape and make them compatible 
+// with our MLIR code. Future work should make this function perform more checks, 
+// we probably want to take the Kernel MLIR argument Types as input and use that 
+// to validate that the passed arguments are good to go. 
+inline py::args simplifiedValidateInputArguments(py::args &args) {
+  py::args processed = py::tuple(args.size());
+  for (std::size_t i = 0; i < args.size(); ++i) {
+    auto arg = args[i];
+    // Check if it has tolist, so it might be a 1d buffer (array / numpy
+    // ndarray)
+    if (py::hasattr(args[i], "tolist")) {
+      // This is a valid ndarray if it has tolist and shape
+      if (!py::hasattr(args[i], "shape"))
+        throw std::runtime_error(
+            "Invalid input argument type, could not get shape of array.");
+
+      // This is an ndarray with tolist() and shape attributes
+      // get the shape and check its size
+      auto shape = args[i].attr("shape").cast<py::tuple>();
+      if (shape.size() != 1)
+        throw std::runtime_error("Cannot pass ndarray with shape != (N,).");
+
+      arg = args[i].attr("tolist")();
+    }
+
+    processed[i] = arg;
+  }
+
+  return processed;
+}
+
 /// @brief For general function broadcasting over many argument
 /// sets, this function will create those argument sets from
 /// the input `args`.
```

Reply:
Fixed in 6225c74